backend/open_webui/main.py

```diff
@@ -4,383 +4,153 @@
 import logging
 import mimetypes
 import os
+import random
 import shutil
 import sys
 import time
-import random
-
 from contextlib import asynccontextmanager
-from urllib.parse import urlencode, parse_qs, urlparse
-from pydantic import BaseModel
-from sqlalchemy import text
-
 from typing import Optional
-from aiocache import cached
+from urllib.parse import parse_qs, urlencode, urlparse
+
 import aiohttp
 import requests
-
-
-from fastapi import (
-    Depends,
-    FastAPI,
-    File,
-    Form,
-    HTTPException,
-    Request,
-    UploadFile,
-    status,
-    applications,
-    BackgroundTasks,
-)
-
-from fastapi.openapi.docs import get_swagger_ui_html
-
+from aiocache import cached
+from fastapi import (BackgroundTasks, Depends, FastAPI, File, Form,
+                     HTTPException, Request, UploadFile, applications, status)
 from fastapi.middleware.cors import CORSMiddleware
+from fastapi.openapi.docs import get_swagger_ui_html
 from fastapi.responses import JSONResponse, RedirectResponse
 from fastapi.staticfiles import StaticFiles
-
-from starlette.exceptions import HTTPException as StarletteHTTPException
-from starlette.middleware.base import BaseHTTPMiddleware
-from starlette.middleware.sessions import SessionMiddleware
-from starlette.responses import Response, StreamingResponse
-
-
-from open_webui.utils import logger
-from open_webui.utils.audit import AuditLevel, AuditLoggingMiddleware
-from open_webui.utils.logger import start_logger
-from open_webui.socket.main import (
-    app as socket_app,
-    periodic_usage_pool_cleanup,
-)
-from open_webui.routers import (
-    audio,
-    images,
-    ollama,
-    openai,
-    retrieval,
-    pipelines,
-    tasks,
-    auths,
-    channels,
-    chats,
-    folders,
-    configs,
-    groups,
-    files,
-    functions,
-    memories,
-    models,
-    knowledge,
-    prompts,
-    evaluations,
-    tools,
-    users,
-    utils,
-)
-
-from open_webui.routers.retrieval import (
-    get_embedding_function,
-    get_ef,
-    get_rf,
-)
-
+from open_webui.config import (  # Ollama; OpenAI; Direct Connections; Tool Server Configs; Code Execution; Image; Audio; Retrieval; Retrieval (Web Search); WebUI; WebUI (OAuth); WebUI (LDAP); Misc; Admin; Tasks
```

Comment:
Should we use the same code formatter as Open WebUI to reduce cost in case we need to sync with upstream in future?

The below command can be used for formatting backend code:

```
npm run format:backend
```

Reply:
sure

Reply:
it's the formatted result, seems the formatter is not that consistent